index.bs

```diff
@@ -18,8 +19,9 @@ Adding Stylesheets In Script {#adding-stylesheets}
 =================================
 
 <pre class='idl'>
-[Constructor(DOMString text, optional CSSStyleSheetInit options)]
-partial interface CSSStyleSheet {
+partial interface Document {
+	[NewObject] Promise&lt;CSSStyleSheet> createCSSStyleSheet(DOMString text, optional CSSStyleSheetInit options);
+	[NewObject] CSSStyleSheet createEmptyCSSStyleSheet(optional CSSStyleSheetInit options);
```

Comment:
Keep the constructor and give it this empty-stylesheet functionality; it doesn't need to be a factory method. In general we should always provide constructors if possible; it's more natural JS style.

Reply:
The reason we wanted it to be a factory method was to associate the stylesheet with the document so that URL and fetch group is clear (see https://github.com/WICG/construct-stylesheets/issues/10#issuecomment-402354965, https://github.com/WICG/construct-stylesheets/issues/15#issuecomment-402952304). Maybe we can do that without putting it in `Document` but I think it's better to have it be clear like this?

Reply:
Oh, I didn't notice the move from CSSStyleSheet to Document! Hm, okay then, carry on.